### DISCUSSION TITLE
Option to generate property specific events

### DISCUSSION THREAD

**Discussion by caesay** [canton7/PropertyChanged.SourceGenerator#31] [Ideas]
I often want to expose just a single property as an event. For example:

```cs
public string FirstName { get/set }
public event EventHandler<PropertyChangedEventArgs<string>> FirstNameChanged;

private void OnFirstNameChanged()
{
    FirstNameChanged?.Invoke(this, new PropertyChangedEventArgs(FirstName));
}
```

This allows WPF binding to work on all properties via INotifyPropertyChanged but allows my own back-end code to just subscribe to the only event it needs. This avoids me needing to do a string comparison on the property name if all I care about is one property.

I am not suggesting that this should be the default, but it would be awesome if this was an option via the `[Notify]` attribute!

**Comment by canton7** [comment]
I normally use something like this: https://github.com/canton7/Stylet/blob/master/Stylet/PropertyChangedExtensions.cs

That lets you do:

```cs
someViewModel.Bind(x => x.FirstName, (o, e) => ...);
```

That subscribes to the PropertyChanged event, filtering on when the PropertyName is "FirstName".
